src/main/scala/dotty/dokka/tasty/ClassLikeSupport.scala

```diff
@@ -35,9 +34,19 @@ trait ClassLikeSupport:
       signatureOnly: Boolean = false,
       modifiers: Seq[Modifier] = classDef.symbol.getExtraModifiers(),
     ): DClass = 
-      val supertypes = getSupertypes(classDef).map{ case (symbol, tpe) =>
-        LinkToType(tpe.dokkaType.asSignature, symbol.dri, kindForClasslike(symbol))
+      def hackGetParents(classDef: ClassDef): Option[List[Tree]] = scala.util.Try(classDef.parents).toOption
+
+      def getSupertypesGraph(classDef: ClassDef, link: LinkToType): Seq[(LinkToType, LinkToType)] = {
+        val smbl = classDef.symbol
+        val parents = if smbl.exists then hackGetParents(smbl.tree.asInstanceOf[ClassDef]) else None
+        parents.fold(Seq())(_.flatMap { case tree =>
+            val symbol = if tree.symbol.isClassConstructor then tree.symbol.owner else tree.symbol
+            val superLink = LinkToType(tree.dokkaType.asSignature, symbol.dri, kindForClasslike(symbol))
+            Seq(link -> superLink) ++ getSupertypesGraph(tree.asInstanceOf[ClassDef], superLink)
+          }       
+        )
       }
+      val supertypes: Map[LinkToType, Seq[LinkToType]] = getSupertypesGraph(classDef, null).groupMap(_._1)(_._2)
```

Comment:
I am not a big fun of `null` and `_1` amd `_2` in this line

I've just learn that we've got `groupMap` :+1: :)

Reply:
I don't understand this `null` either. Am I missing something, or is `null` the only key that we ever put in the map? Because if that's the case, then we shouldn't have a `Map` at all.

Reply:
`null` value is a trick here. The question is: Is it possible to get `LinkToType` or `tree` from just `ClassDef`.
If the answer is yes, tell me how, and I will fix it.
If the answer is no, let me explain how it works.
As I construct the graph as an adjacency list I need to identify each node. From `ClassLikeSupport`, I cannot obtain proper `LinkToType` element for a root element. However, I can use `null` value as the ID for such case. Later, in hierarchy diagram builder, I can identify such element, and associate him a proper `LinkToType`, because at this step I have access to `Member.asLink` extension function

Reply:
Ah, okay, I see - I'm just blind, there's a recursive call to `getSupertypesGraph`. Is calculating the graph lazily, after we already have the `Member` object an option? If yes, then we can do that and call `.asLink` there. Otherwise I think the nice Scala way would be to define an enum like:
```
enum GraphKey:
  case Root
  case Linked(link: LinkToType)
```
and use that as the key.